src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/university/TryResearchMessage.java

```diff
@@ -76,50 +96,132 @@ protected void onExecute(
             return;
         }
 
+        final IGlobalResearch research = IGlobalResearchTree.getInstance().getResearch(branch, researchId);
+        if(reset)
+        {
+            if(colony.getResearchManager().getResearchTree().getResearch(branch, researchId) != null)
```

Comment:
I've added a new sound handler thing for success and failure sounds, would make sense to integrate this here